tools/refengine/root.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2017 oci-discovery contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package refengine
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+)
+
+// MerkleRoot holds a single resolved Merkle root.
+type MerkleRoot struct {
+
+	// The Merkle root object.  While this may be of any type.  OCI
+	// tools will generally use image-spec Descriptors.
+	Root interface{}
```

Comment:
We may want a `MediaType string` property on `MerkleRoot` so we don't lose typing information in any JSON output.  We still have Go typing information without it, because callers can use reflection and/or casting to determine if `Root` is backed by a type they understand, but having an explicit media type stores that information in a way that will survive Go serialization.  I can add that to this commit, or make it part of follow-up work.

Reply:
> We may want a `MediaType` string property…

Added with 9529456 → bb96e49.